packages/cosmic-swingset/lib/ag-solo/vats/weakStore.js

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2019 Agoric, under Apache license 2.0
+
+import harden from '@agoric/harden';
+import { insist } from '@agoric/ertp/util/insist';
+/**
+ * Distinguishes between adding a new key (init) and updating or
+ * referencing a key (get, set, delete).
+ *
+ * `init` is only allowed if the key does not already exist. `Get`,
+ * `set` and `delete` are only allowed if the key does already exist.
+ * @param  {string} keyName - the column name for the key
+ */
+function makeStore(keyName = 'key') {
```

Comment:
Don't we need only one copy of this?

Reply:
One's a weakMap wrapper and one's a Map wrapper.